sys/fido2/ctap/ctap.c

```diff
@@ -0,0 +1,1508 @@
+/*
+ * Copyright (C) 2021 Nils Ollrogge
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup fido2_ctap
+ * @{
+ * @file
+ *
+ * @author  Nils Ollrogge <nils-ollrogge@outlook.de>
+ * @}
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
+
+#include "fmt.h"
+#include "xtimer.h"
+#include "byteorder.h"
+
+#include "fido2/ctap/transport/ctap_transport.h"
+#include "fido2/ctap.h"
+#include "fido2/ctap/ctap_utils.h"
+#include "fido2/ctap/ctap_cbor.h"
+#include "fido2/ctap/ctap_mem.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/*** main CTAP2 functions ***/
+
+/**
+ * @brief MakeCredential method
+ *
+ * CTAP specification (version 20190130) section 5.1
```

Comment:
CTAP is correct. It is the CTAP specification of FIDO version 2.

What do you mean by filename and folders ? Do you mean that the ctap.c file should be moved out of the ctap folder ?

Reply:
Sorry missed some strings.

I left the status codes because they are explicitly defined like this in the [spec](https://fidoalliance.org/specs/fido-v2.0-ps-20190130/fido-client-to-authenticator-protocol-v2.0-ps-20190130.html#error-responses).

Reply:
Well you can look at it in two ways. The CTAP spec I implemented is an evolution of a previous spec, therefore one could call it CTAP2. On the other hand, the heading of the spec says CTAP not CTAP2. 

I personally would leave it at CTAP but if you think CTAP2 makes more sense I can revert back to ctap2.

Reply:
From [section 1.1 of the specification](https://fidoalliance.org/specs/fido-v2.0-ps-20190130/fido-client-to-authenticator-protocol-v2.0-ps-20190130.html#relationship-to-other-specifications):

> Occasionally, the term "CTAP" may be used without clarifying whether it is referring to CTAP1 or CTAP2. In such cases, it should be understood to be referring to the entirety of this specification or portions of this specification that are not specific to either CTAP1 or CTAP2. For example, some error messages begin with the term "CTAP" without clarifying whether they are CTAP1- or CTAP2-specific because they are applicable to both CTAP protocol versions. CTAP protocol-specific error messages are prefixed with either "CTAP1" or "CTAP2" as appropriate.

It seems to me, that if the whole specification is being implemented here, we can use `ctap`, as it is "referring to the entirety of this specification". Documentation will further clarify what the module is implementing anyway.